drivers/ieee802154/ieee802154_cc2520.c

```diff
@@ -973,6 +963,84 @@ static int power_on_and_setup(struct device *dev)
 	return 0;
 }
 
+static struct cc2520_gpio_configuration *configure_gpios(struct device *dev)
+{
+	struct cc2520_context *cc2520 = dev->driver_data;
+	const int flags_noint_out = GPIO_OUTPUT;
+	const int flags_noint_in = GPIO_INPUT;
+	const int flags_int_in = (GPIO_INPUT | DT_CC2520_INT_FLAGS);
+	struct device *gpio;
+
+	/* VREG_EN */
+	gpio = device_get_binding(DT_CC2520_GPIO_VREG_EN_DRV_NAME);
+	if (!gpio) {
+		return NULL;
+	}
+
+	cc2520->gpios[CC2520_GPIO_IDX_VREG_EN].pin = DT_CC2520_GPIO_VREG_EN;
+	gpio_pin_configure(gpio, cc2520->gpios[CC2520_GPIO_IDX_VREG_EN].pin,
+			   flags_noint_out);
```

Comment:
All pin configuration must incorporate the devicetree flags along with the direction flags.